src/pkg/projects/packaging-tools/windows/provider.wxs

```diff
@@ -2,7 +2,7 @@
 <Wix xmlns="http://schemas.microsoft.com/wix/2006/wi" xmlns:dep="http://schemas.microsoft.com/wix/DependencyExtension">
   <?include "Variables.wxi" ?>
   <Fragment>
-    <Component Id="$(var.DependencyKeyId)" Directory="TARGETDIR" Win64="no" Guid="117B1755-5027-4257-A1A9-7C37D81C3D5F">
+    <Component Id="$(var.DependencyKeyId)" Directory="TARGETDIR" Win64="no" Guid="$(var.DependencyKeyGuid)">
```

Comment:
What is the functional purpose of the "var.DependencyKeyId" GUID?

Reply:
(Took a journey for this one... turns out there's been a bug here that I can fix! 😄)

In the scope of this PR, I made this change so that both the .NET Core and Windows Desktop targeting packs can use the same `wxs` file but have unique GUIDs here.

From the ground up, I don't know why this `Component` has a project-specific GUID. I saw that each existing `provider.wxs` in Core-Setup has a unique GUID in the `Guid` attribute, and I've been keeping to that as the status quo.

Now... if I remove `Guid="$(var.DependencyKeyGuid)"` (to autogenerate), the build succeeds, and I see that the provider component's GUID is some generated value. I think this is ok, because this component puts down a creates a version-specific registry key (`Software\Classes\Installer\Dependencies\dotnet_targeting_pack_24.0.27515_x64`) that won't be shared with future builds. In fact, reading general MSI guidance, using a static component GUID is really bad when the artifact is being renamed: this messes up ref counting.

Looking at [Core-SDK's `provider.wxs`](https://github.com/dotnet/core-sdk/blob/master/src/redist/targets/packaging/windows/clisdk/provider.wxs), it explicitly uses the auto-GUID. In fact, tracing it back, the static GUID was removed in CLI's `provider.wxs` to fix a bug where registry entries are left behind: https://github.com/dotnet/core-setup/issues/40#issuecomment-225323170.

It turns out that that bug affects the Core-Setup MSIs: the reg key stays behind when I install and uninstall (for example) the host MSI.

I'm going to add a change that makes all `provider.wxs`s in Core-Setup auto-generate these GUIDs to fix that.

@johnbeisner let me know if that train of thought doesn't make sense, thanks for pointing this out!